File: Frontend/projectk-frontend/src/app/features/kurinModule/common/models/memberWarningDto.ts
import { MemberWarningLevel } from './enums/member-warning-level.enum';

export interface MemberWarningDto {
  memberWarningKey: string;
  memberKey: string;
  level: MemberWarningLevel;
  issuedAtUtc: string;
  expiresAtUtc: string;
  issuedByUserKey: string;
  revokedByUserKey?: string | null;
  revokedAtUtc?: string | null;
}

File: Backend/ProjectK.Backend/ProjectK.Common/Models/Dtos/MemberWarningDto.cs
using ProjectK.Common.Models.Enums;
using System;

namespace ProjectK.Common.Models.Dtos
{
    public class MemberWarningDto
    {
        public Guid MemberWarningKey { get; set; }
        public Guid MemberKey { get; set; }
        public MemberWarningLevel Level { get; set; }
        public DateTime IssuedAtUtc { get; set; }
        public DateTime ExpiresAtUtc { get; set; }
        public Guid IssuedByUserKey { get; set; }
        public Guid? RevokedByUserKey { get; set; }
        public DateTime? RevokedAtUtc { get; set; }
    }
}
